packages/core/src/render3/query.ts

```diff
@@ -111,14 +107,24 @@ function add(predicate: QueryPredicate<any>| null, node: LNode) {
           predicate.values.push(directives[i << 1]);
         }
       }
+    } else {
+      const staticData = node.staticData;
+      if (staticData) {
+        const selector = predicate.selector !;
+        for (var i = 0; i < selector.length; i++) {
+          if (selector[i] === staticData.localName) {
+            predicate.values.push(new ElementRef(node.native));
```

Comment:
We should be getting `ElementRef` from the DI instead. Need to discuss the `injectElementRef()` and similar functions first.

Reply:
Yes, we should use DI, What is the question? You will need something like `injectElementRef` but you will have to pass in node explicitly somehow into it.